flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -91,17 +102,17 @@ private void start() {
             } else {
                 libDirs = Collections.emptyList();
             }
-            final Executor executor = new LocalExecutor(options.getDefaults(), jars, libDirs);
+            final Executor executor = new LocalExecutor();
             executor.start();
 
-            // create CLI client with session environment
-            final Environment sessionEnv = readSessionEnvironment(options.getEnvironment());
-            appendPythonConfig(sessionEnv, options.getPythonConfiguration());
+            // build default context
+            DefaultContext defaultContext = buildEmbeddedDefaultContext(jars, libDirs);
```

Comment:
Personally, I think `buildEmbeddedDefaultContext` should belong to `LocalExecutor`. `SqlClient` shouldn't care about how to build context, it just need to setup Executor (`LocalExecutor`, or `GatewayExecutor` in the future). 

I think we can also improve the interface of `Executor`, changing `openSession` to `openSession(String sessionId)` and create `SessionContext` in it.